### ISSUE TITLE
Consider an Ensure.Dictionary type-specific class

### ISSUE THREAD

**Issue by ndrwrbgs** [danielwertheim/Ensure.That#53]
Problem: Intellisense shows the IDictionary methods when I do Ensure.Collection, which are not valid on an `ICollection`. Could we introduce a new `DictionaryArg` class as a subclass of `CollectionArg`?

**Comment by danielwertheim** [comment]
Is there a real problem or semantic issue? The idea was that "Collection'ish" stuff should be grouped there under and not have one for Sets, Dictionaries.... as e.g. both those extens ICollection<T>

**Comment by ndrwrbgs** [comment]
There's no problem, it's usable in either scenario. What I was hoping for was intellisense being able to show me everything I can do based on my argument type (whether inferred from `Ensure.That.X()` or specified by me via `Ensure.Type.X()`). When I was using it I was briefly confused that for `Ensure.Collection` it showed me methods that only accepted Dictionary, since I was trying to validate an argument that was passed in as `ICollection` and so those methods wouldn't be applicable (but I wouldn't see this until I try to use it and the IDE says the argument type isn't compatible).

TL;DR it's just an enhancement suggestion and definitely 💯% NOT a blocking change/issue.

**Comment by ndrwrbgs** [comment]
v8 Breaking change

**Comment by danielwertheim** [comment]
I'm not keen on creating one per type `Array, List, Set, Collection, Dictionary` at least not for now. Hence I don't think it's correct to let Dictionary exist all by it self. `Ensure.Collection` is to be seen as everything collection'ish as in stuff under the `System.Collections` namespace.

@ndrwrbgs  If you do believe this will become a technical problem seen to API overloads, feel free to re-open, and then this will be implemented.

**Comment by ndrwrbgs** [comment]
Nope it was purely for discoverability of the methods not for conflicts, also no reason you’d have to create all - most of those other classes don’t have special methods that only take them (most of the methods take ICollection/enumerable so would be fine to stay in EmumerableMethods, only the dictionary ones deviate)

Anyway, just wanted to reply but am fine with it being closed too
